keras_cv/layers/preprocessing/erasing.py

```diff
@@ -0,0 +1,196 @@
+import abc
+
+import tensorflow as tf
+import tensorflow.keras.layers as layers
+from tensorflow.python.keras.utils import layer_utils
+
+from keras_cv.utils import fill_utils
+
+
+class BaseErasing(layers.Layer, abc.ABC):
+    """This can be inherited by layers that wants to implement erasing of patches.
+
+    Args:
+        rate: Float between 0 and 1.  The fraction of samples to augment.
+        patch_value: Float or string. The value to fill in the patches. If "gaussian", will
+            fill patches with gaussian noise. Defaults to "gaussian".
+    """
+
+    def __init__(self, rate, patch_value="gaussian", seed=None, **kwargs):
+        super().__init__(**kwargs)
+
+        if isinstance(patch_value, str):
+            layer_utils.validate_string_arg(
+                patch_value,
+                allowable_strings=["gaussian"],
+                layer_name=self.__class__.__name__,
+                arg_name="patch_value",
+                allow_none=False,
+                allow_callables=False,
+            )
+
+        self.rate = rate
+        self.patch_value = patch_value
+        self.seed = seed
+
+    def call(self, images, labels):
```

Comment:
All augmentation layers should have a `training=True` argument. When False (inference mode) no augmentation is applied.